utils/src/server/rpc_module.rs

```diff
@@ -406,12 +406,13 @@ impl<Context> DerefMut for RpcModule<Context> {
 pub struct RpcModule<Context> {
 	ctx: Arc<Context>,
 	methods: Methods,
+	notif_overrides: SubscriptionNotifOverride,
 }
 
 impl<Context> RpcModule<Context> {
 	/// Create a new module with a given shared `Context`.
 	pub fn new(ctx: Context) -> Self {
-		Self { ctx: Arc::new(ctx), methods: Default::default() }
+		Self { ctx: Arc::new(ctx), methods: Default::default(), notif_overrides: SubscriptionNotifOverride::default() }
```

Comment:
This doesn't really prevent us from having the same notification name in two different modules (the set is local to the instance and is thrown away the moment you turn it into `Methods`).

If we can't easily guarantee that notification names are unique, and there isn't really much of a point in checking that they are in the server (notifications will be sent regardless, and we rely on sub id more than name anyway), I think it might actually be better to drop the check altogether.

Reply:
ty, can't we handle it in `RpcModule::merge`?

Reply:
You can, but you'd have to move the hashset inside `Methods` because that's what merge operates on.

Reply:
let's keep it simple I removed all of this but kept the check in the proc macros for now, ok?

Reply:
Good pt @maciejhirsz. I'm ok making this a "sharp" feature, but we need to document it properly. "NOTE: the author is expected to ensure the uniqueness of the override across **all** `RpcModule`s used on the server." or some such.

Reply:
Note, we have a check for this in the `proc macros` where each trait ensures uniqueness for each name.

Do you want me to remove it to somewhat consistent?